File: languages/javascript/templates/declarations/event.js
  /**
   * ${method.summary}
   * 
   * @param {'${event.name}'} event
   * @param {Function} callback
${if.deprecated}   * @deprecated ${method.deprecation}
${end.if.deprecated}  */
  function listen(event: '${event.name}'${if.context}, ${event.signature.params}${end.if.context}, callback: (data: ${method.result.type}) => void): Promise<number>

  /**
   * ${method.summary}
   * When using `once` the callback method will only fire once, and then disconnect your listener
   * 
   * @param {'${event.name}'} event
   * @param {Function} callback
${if.deprecated}   * @deprecated ${method.deprecation}
${end.if.deprecated}  */
function once(event: '${event.name}'${if.context}, ${event.signature.params}${end.if.context}, callback: (data: ${method.result.type}) => void): Promise<number>

File: languages/c/templates/declarations/event.c
/* ${method.name} - ${method.description} */
typedef void (*F${info.Title}${method.Name}Callback)( const void* userData, ${event.signature.callback.params}${if.event.params}, ${end.if.event.params}${method.result.properties} );
int32_t F${info.title}_Register_${method.Name}( ${event.signature.params}${if.event.params}, ${end.if.event.params}F${info.Title}${method.Name}Callback userCB, const void* userData );
int32_t F${info.title}_Unregister_${method.Name}( F${info.Title}${method.Name}Callback userCB);

// signature callback params: ${event.signature.callback.params}
// method result properties : ${method.result.properties}
